src/AugmentedMatrix.jl

```diff
@@ -0,0 +1,62 @@
+using LinearAlgebra
+
+#Used in one line in function below
+function invert_or_zero(x :: Real)
+    if x == 0
+        return 0
+    else
+        return 1/x
+    end
+end
+
+#Code to compute augmented rate matrix
+function augmentedRateMatrix(list_of_rate_matrices :: Array{B}, list_of_time_steps :: Array{T}) where {T <: Real, B <: Matrix{T}}
+    number_of_times = length(list_of_time_steps)
+    @assert length(list_of_rate_matrices) == number_of_times
+    @assert length(list_of_rate_matrices) ≥ 1
```

Comment:
Using `@assert` to check conditions of the arguments is usually wrong. According to the macro's documentation, `@assert` clauses may be removed when certain optimizations are turned on, so they should only be used for debugging, not error checking in production